lib/improver/tests/standardise/test_StandardiseGridAndMetadata.py

```diff
@@ -0,0 +1,301 @@
+            self.cube, new_name=new_name, new_units="degC",
+            coords_to_remove=["forecast_period"],
+            attributes_dict=attribute_changes)
+        self.assertEqual(result.name(), new_name)
+        self.assertEqual(result.units, "degC")
+        self.assertArrayAlmostEqual(result.data, expected_data, decimal=5)
+        self.assertDictEqual(result.attributes, expected_attributes)
+        self.assertNotIn(
+            "forecast_period", [coord.name() for coord in result.coords()])
+
+    def test_fix_float64(self):
+        """Test precision de-escalation"""
+        self.cube.data = self.cube.data.astype(np.float64)
+        result = self.plugin.process(self.cube, fix_float64=True)
+        self.assertEqual(result.data.dtype, np.float32)
+
+
+class Test_process_regrid_options(IrisTest):
+    """Test the process method with regridding options. Regridded values
+    are not tested here as this is covered by unit tests for the regridding
+    routines (iris.cube.Cube.regrid and improver.standardise.RegridLandAndSea).
+    """
+
+    def setUp(self):
+        """Set up input cubes and landmask"""
+        self.cube = set_up_variable_cube(
+            282*np.ones((15, 15), dtype=np.float32), spatial_grid='latlon',
+            standard_grid_metadata='gl_det')
+
+        # set up dummy landmask on source grid
+        landmask = np.full((15, 15), False)
+        self.landmask = self.cube.copy(data=landmask)
+        self.landmask.rename("land_binary_mask")
+        self.landmask.units = "no_unit"
+        for coord in ["time", "forecast_reference_time", "forecast_period"]:
+            self.landmask.remove_coord(coord)
+
+        # set up dummy landmask on high resolution target grid
+        self.target_grid = set_up_variable_cube(
+            np.full((12, 12), False), name="land_binary_mask", units="no_unit",
+            spatial_grid='equalarea', standard_grid_metadata='uk_det')
+        for coord in ["time", "forecast_reference_time", "forecast_period"]:
+            self.target_grid.remove_coord(coord)
+
+    def test_basic_regrid(self):
+        """Test default regridding arguments return expected dimensionality
+        and updated grid-defining attributes"""
+        expected_data = 282*np.ones((12, 12), dtype=np.float32)
+        expected_attributes = {"mosg__model_configuration": "gl_det"}
+        for attr in ["mosg__grid_domain", "mosg__grid_type",
+                     "mosg__grid_version"]:
+            expected_attributes[attr] = self.target_grid.attributes[attr]
+        result = StandardiseGridAndMetadata().process(
+            self.cube, self.target_grid.copy())
+        self.assertArrayAlmostEqual(result.data, expected_data)
+        for axis in ['x', 'y']:
+            self.assertEqual(
+                result.coord(axis=axis), self.target_grid.coord(axis=axis))
+        self.assertDictEqual(result.attributes, expected_attributes)
+
+    def test_access_regrid_with_landmask(self):
```

Comment:
I've noticed that this unit test is currently generating a warning of the form below, which seems to coming from the line `cube = cube.regrid(target_grid, regridder)`. This is handled in the `test_AdjustLandSeaPoints` tests using:
```
    # The warning messages are internal to the iris.analysis module v2.2.0.
    @ManageWarnings(ignored_messages=["Using a non-tuple sequence for "],
                    warning_types=[FutureWarning])
```

Note that hiding this warning message in `test_access_regrid_with_landmask` might mean that it pops up for another unit test, so it might be that most of the unit tests in `Test_process_regrid_options` need `ManageWarnings`.

The full warning message is: 
```
FutureWarning: Using a non-tuple sequence for multidimensional indexing is deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`. In the future this will be interpreted as an array index, `arr[np.array(seq)]`, which will result either in an error or a different result.
   return self.values[idx_res]
```

Reply:
So I have done this, but I'm a little concerned we're suppressing a `FutureWarning`.  What plans do we have to fix this issue?